src/routers/UpdateMatchAvailabilitiesRouter.ts

```diff
@@ -0,0 +1,80 @@
+import { Request } from 'express';
+import Constants from '../common/constants';
+import Availability from '../entities/Availability';
+import AvailabilityRepo from '../repos/AvailabilityRepo';
+import MatchRepo from '../repos/MatchRepo';
+import AuthenticatedApplicationRouter from '../utils/AuthenticatedApplicationRouter';
+
+class UpdateMatchAvailabilitiesRouter extends AuthenticatedApplicationRouter<void> {
+  constructor() {
+    super('POST');
+  }
+
+  getPath(): string {
+    return '/availabilities/';
+  }
+
+  async content(req: Request): Promise<void> {
+    const { matchID, schedule } = req.body;
+
+    if (!matchID) throw Error('Missing required matchID');
+    if (!schedule) throw Error('Missing required schedule list');
+
+    // Logic to update match status
+    const match = await MatchRepo.getMatchByID(matchID);
+    let status = Constants.MATCH_CREATED;
+    if (match.status !== Constants.MATCH_CREATED && match.status !== Constants.MATCH_PROPOSED) {
+      throw Error('Cannot update match availability at this stage!');
+    }
+    if (match.status === Constants.MATCH_CREATED) {
+      status = Constants.MATCH_PROPOSED;
+    }
+    if (match.status === Constants.MATCH_PROPOSED) {
```

Comment:
in fact the match.status can only ever be one thing at a time, probably best to make this and if, else if, and the else throws the error